services/tracing/tracing/transport/http.go

```diff
@@ -0,0 +1,71 @@
+package tracingtransport
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"sensorbucket.nl/sensorbucket/internal/httpfilter"
+	"sensorbucket.nl/sensorbucket/internal/pagination"
+	"sensorbucket.nl/sensorbucket/internal/web"
+	"sensorbucket.nl/sensorbucket/services/tracing/tracing"
+)
+
+type HTTPTransport struct {
+	router chi.Router
+	svc    *tracing.Service
+	url    string
+}
+
+func NewHTTP(svc *tracing.Service, url string) *HTTPTransport {
+	t := &HTTPTransport{
+		router: chi.NewRouter(),
+		svc:    svc,
+		url:    url,
+	}
+	t.SetupRoutes(t.router)
+	return t
+}
+
+func (t *HTTPTransport) SetupRoutes(r chi.Router) {
+	r.Get("/health", func(rw http.ResponseWriter, r *http.Request) {
+		rw.Write([]byte("healthy"))
+	})
+	r.Get("/tracing", t.httpGetTraces())
+}
+
+func (t *HTTPTransport) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	t.router.ServeHTTP(w, r)
+}
+
+func (t *HTTPTransport) httpGetTraces() http.HandlerFunc {
+	type Params struct {
+		tracing.Filter     `pagination:",squash"`
+		pagination.Request `pagination:",squash"`
+	}
+	return func(rw http.ResponseWriter, r *http.Request) {
+		params, err := httpfilter.Parse[Params](r)
+		if err != nil {
+			web.HTTPError(rw, web.NewError(http.StatusBadRequest, "invalid params", ""))
+			return
+		}
+
+		if params.DurationGreaterThan != nil &&
+			params.DurationLowerThan != nil &&
+			*params.DurationGreaterThan != 0 &&
+			*params.DurationLowerThan <= *params.DurationLowerThan {
+			web.HTTPError(rw, web.NewError(http.StatusBadRequest, "duration_greater_than cannot be smaller than or equal to duration_smaller_than", ""))
+			return
+		}
```

Comment:
Why check for durationgreathan != 0?

Reply:
yeah it's not relevant. also there is something else wrong with the check now that I look at it:

*params.DurationLowerThan <= *params.DurationLowerThan

Reply:
also, greater the check should be reversed. greater_than SHOULD be smaller than lower then because when both filters are included you only want values between the 2